packages/cli/src/util/dev/server.ts

```diff
@@ -1645,7 +1647,7 @@ export default class DevServer {
       prevUrl =
         routeResult.continue && routeResult.dest
           ? getReqUrl(routeResult)
-          : req.url;
+          : prevUrl;
```

Comment:
Small knit pick, but consider a simple `if` instead of a ternary when the l-value and one of the r-values are the same.

```javascript
if (routeResult.continue && routeResult.dest) {
  prevUrl = getReqUrl(routeResult);
}
```

Reply:
I do like this more, yeah. I realize now that `prevHeaders` would have the same problem. I updated both values. [`de4c550` (#8535)](https://github.com/vercel/vercel/pull/8535/commits/de4c550d5e4f9d8d824a7ebe6fe29f8c2df2d4fe)